nntrainer/graph/network_graph.cpp

```diff
@@ -313,6 +313,12 @@ int NetworkGraph::checkInitializedGraph() {
     if (lnode->getTrainable() ||
         must_support_backwarding.find(lnode->getName()) !=
           must_support_backwarding.end()) {
+      lnode->needsCalcGradient(true);
```

Comment:
If we are going to do this, I think layerNode::calcGradient should not implicitly make calcGradient noop.

below is related snippet.
```cpp
  if (getTrainable())
    layer->calcGradient(*run_context);
```

Reply:
updated  `if (getTrainable())` to `if (needs_calc_gradient)`.